tests/testthat/test-rapporttekst-funksjoner.R

```diff
@@ -113,3 +143,44 @@ test_that("akse_tall_format() gir ut en tom tekstvektor hvis den tar
   tall_format = akse_tall_format()
   expect_identical(tall_format(integer()), character())
 })
+
+# Testing av prosent()
+
+test_that("prosent() gir ut tankestrek når «x»-argumentet inneholder NA", {
+  expect_identical(prosent(NA_real_), "\\textendash{}")
+})
+
+test_that("prosent() gir ut riktig verdi når kun «x»-argumentet er gitt", {
+  expect_identical(prosent(numeric()), character(0))
+  expect_identical(prosent(0.1234), "{\\numprint{12}}\\prosent")
+  expect_identical(
+    prosent(c(0.1234, 0.31415)),
+    c("{\\numprint{12}}\\prosent", "{\\numprint{31}}\\prosent")
+  )
+})
+
+test_that("prosent() gir ut riktig verdi med «desimalar»-argumentet", {
+  expect_identical(
+    prosent(0.1234, desimalar = 0),
+    "{\\numprint{12}}\\prosent"
+  )
+  expect_identical(
+    prosent(0.1234, desimalar = 2),
+    "{\\numprint{12.34}}\\prosent"
+  )
+  expect_identical(
+    prosent(c(0.1234, 0.31), desimalar = 3),
+    c("{\\numprint{12.340}}\\prosent", "{\\numprint{31.000}}\\prosent")
+  )
+})
+
+test_that("prosent() gir error om «x»-argument er i tekstform", {
+  expect_error(
+    prosent("0.12"),
+    "Assertion on 'x' failed: Must be of type 'numeric', not 'character'."
```

Comment:
Her *antar* du noko om korleis checkmate-funksjonen skal fungera og nøyaktig formulering av feilmeldinga. Det har du ingenting med! Det er det berre checkmate-funksjonen sjølv som skal halda orden på.

Kvar funksjonen har ansvar for si eiga åtferd. Bruk heller:

```r
checkmate::check_numeric("0.12")
```

for å få ut feilmeldinga som checkmate-funksjonen skal gje ut når inndata er tekst men skal vera numerisk (denne feilmeldinga kan godt endrast i framtidige versjonar av checkmate).